packages/anoncreds/src/anoncreds-rs/AnonCredsRsHolderService.ts

```diff
@@ -199,21 +200,52 @@ export class AnonCredsRsHolderService implements AnonCredsHolderService {
       }
 
       const credentialsProve: CredentialProve[] = []
-      const credentials: { linkSecretId: string; credentialEntry: CredentialEntry }[] = []
+      const credentials: { linkSecretId: string; credentialEntry: CredentialEntry; credentialId: string }[] = []
 
       let entryIndex = 0
       for (const referent in selectedCredentials.attributes) {
         const attribute = selectedCredentials.attributes[referent]
-        credentials.push(await credentialEntryFromAttribute(attribute))
-        credentialsProve.push({ entryIndex, isPredicate: false, referent, reveal: attribute.revealed })
-        entryIndex = entryIndex + 1
+
+        // If the credentialId with the same timestamp is already present, we will use the existing entry, so that the proof is created
+        // showing the attributes come from the same cred, rather than different ones.
+        const existingCredentialIndex = credentials.findIndex(
+          (credential) =>
+            credential.credentialId === attribute.credentialId &&
+            attribute.timestamp === credential.credentialEntry.timestamp
```

Comment:
Won't the `credentialId` suffice? Why is the timestamp included?

Reply:
I wasn't 100 sure, but as we create the revocation state based on the timestamp, it would be possible to have different requirements.

E.g. age > 15 nonRevoked between 100 and 1000 (timestamp)
and name = Berend. nonRevoked between 10.000 and 100.000 (timestamp)..

These would create different revocation states, and therefore I decided to also include timestamp. It's very very rare to have different timestamps, but this at least makes sure the presentation is valid

Reply:
Ah okay, makes sense! good catch!